torch/csrc/jit/interpreter.cpp

```diff
@@ -831,8 +831,10 @@ struct InterpreterStateImpl : c10::intrusive_ptr_target {
   c10::intrusive_ptr<Future> future;
   std::shared_ptr<CodeImpl> function; // keep function alive
   // these are just copies of function to prevent indirections in interpreter
-  int* int_data;
-  const std::vector<bool>& bool_data;
+  int* int_data_;
+  int* int_data() {return int_data_; }
+  const std::vector<bool>& bool_data_;
+  const std::vector<bool>& bool_data() { return bool_data_; }
```

Comment:
y tho

Reply:
it gives us some flexibility to experiment with different implementations of `bool_data()`, `registers()`, `int_data()` in the presence of `Frame`s in Interpreter as it seems to neatly encapsulate an implementation.
For instance, a very nice implementation albeit not the most efficient one could be
```cpp
int* int_data() {
  return frames_[frames_.size()-1].int_data_
}
```
It could also be something like 
```cpp
int* int_data() {
  return int_data_ + frames_[frames_.size()-1].int_data_offset_;
}
```
or every time we push a frame we could be simply updating Interpreter's `int_data_` to avoid any indirection involving `frames_[frames_.size()-1]` when we call `int_data()`

@zdevito was also suggesting considering reserving space on `Stack` for registers, so the implementation for 

`IValue* registers()` could be 

```cpp
IValue* registers() {
  return stack + frames_[frames_size()-1].registers_start;
}
```

This is just a starting point for a discussion